docker/daemon/daemon_dest.go

```diff
@@ -22,36 +17,49 @@ import (
 
 type daemonImageDestination struct {
 	ref daemonReference
-	// For talking to imageLoadGoroutine
+	img *dockerArchive.Image
+
+	// Related to the goroutine which runs the `docker load`.
 	goroutineCancel context.CancelFunc
 	statusChannel   <-chan error
-	writer          *io.PipeWriter
-	tar             *tar.Writer
-	// Other state
-	committed bool // writer has been closed
+	committed       bool
+
+	// Horrible hack.
+	pipeW *io.PipeWriter
+	pipeR *io.PipeReader
```

Comment:
No single-letter variables, or single-letter-word-abbreviations, in file-scope data structure, please. Let’s keep our brain capacity available for understanding the code instead of just trying to remember what a letter means :)